src/main/java/com/uid2/client/PublisherUid2Helper.java

```diff
@@ -0,0 +1,112 @@
+package com.uid2.client;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.time.Instant;
+import java.util.Arrays;
+
+
+public class PublisherUid2Helper {
+    public PublisherUid2Helper(String base64SecretKey) {
+        secretKey = InputUtil.base64ToByteArray(base64SecretKey);
+    }
+
+    public Envelope createEnvelope(IdentityInput identityInput) {
+        final int nonceLength = 8;
+        byte[] nonce = new byte[nonceLength];
+        secureRandom.nextBytes(nonce);
+        return createEnvelopeImpl(identityInput, nonce, Instant.now(), null);
+    }
+
+    public IdentityTokens createIdentityfromTokenGenerateResponse(String response, byte[] nonce) {
```

Comment:
Should we take `nonce` as argument or should we require passing in the whole request envelope? The latter would abstract away how exactly we check response against the request.

Reply:
I considered both, decided on `nonce` because it can be better to provide functions with only what they need. But on reflection, abstracting outweighs that so will change it.